Added date filtering option and date sorting option

rpeyron opened the pull request.

Scarygami commented on location_history_json_converter.py:
```diff
@@ -56,6 +73,11 @@ def main(argv):
+        
+        if args.chronological:
+            if args.startdate or args.enddate:
+                items = [ item for item in items if dateCheck(item["timestampMs"]) ]
```
I think the filtering for `startdate`, `enddate` should be moved out of the `args.chronological` branch, which would have the advantage that all the `if not dateCheck(item["timestampMs"])  :  continue` lines can be removed

Scarygami commented on location_history_json_converter.py:
```diff
@@ -67,6 +89,7 @@ def main(argv):
             first = True
 
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Could be removed as per comment above.

Scarygami commented on location_history_json_converter.py:
```diff
@@ -84,6 +107,7 @@ def main(argv):
         if args.format == "csv":
             f_out.write("Time,Location\n")
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Could be removed as per comment above.

Scarygami commented on location_history_json_converter.py:
```diff
@@ -126,6 +151,7 @@ def main(argv):
             f_out.write("  </metadata>\n")
             if args.format == "gpx":
                 for item in items:
+                    if not dateCheck(item["timestampMs"]) :  continue
```
Could be removed as per comment above.

Scarygami commented on location_history_json_converter.py:
```diff
@@ -94,6 +118,7 @@ def main(argv):
             f_out.write("  <Document>\n")
             f_out.write("    <name>Location History</name>\n")
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Could be removed as per comment above.

Scarygami commented on location_history_json_converter.py:
```diff
@@ -149,8 +175,9 @@ def main(argv):
                 # The deltas below assume input is in reverse chronological order.  If it's not, uncomment this:
                 # items = sorted(data["data"]["items"], key=lambda x: x['timestampMs'], reverse=True)
                 for item in items:
+                    if not dateCheck(item["timestampMs"]) :  continue
```
Could be removed as per comment above.

askariikeka commented:
What is this ? 

Sent from my iPhone

> On Sep 19, 2016, at 9:41 AM, Gerwin Sturm <notifications@github.com> wrote:
> 
> @Scarygami requested changes on this pull request.
> 
> Thanks for the PR. Will merge once you made the suggested changes.
> 
> In location_history_json_converter.py:
> 
> > @@ -56,6 +73,11 @@ def main(argv):
>              return
>  
>          items = data["locations"]
> +        
> +        if args.chronological:
> +            if args.startdate or args.enddate:
> +                items = [ item for item in items if dateCheck(item["timestampMs"]) ]
> I think the filtering for startdate, enddate should be moved out of the args.chronological branch, which would have the advantage that all the if not dateCheck(item["timestampMs"]) : continue lines can be removed
> 
> In location_history_json_converter.py:
> 
> > @@ -67,6 +89,7 @@ def main(argv):
>              first = True
>  
>              for item in items:
> +                if not dateCheck(item["timestampMs"]) :  continue
> Could be removed as per comment above.
> 
> In location_history_json_converter.py:
> 
> > @@ -84,6 +107,7 @@ def main(argv):
>          if args.format == "csv":
>              f_out.write("Time,Location\n")
>              for item in items:
> +                if not dateCheck(item["timestampMs"]) :  continue
> Could be removed as per comment above.
> 
> In location_history_json_converter.py:
> 
> > @@ -126,6 +151,7 @@ def main(argv):
>              f_out.write("  </metadata>\n")
>              if args.format == "gpx":
>                  for item in items:
> +                    if not dateCheck(item["timestampMs"]) :  continue
> Could be removed as per comment above.
> 
> In location_history_json_converter.py:
> 
> > @@ -94,6 +118,7 @@ def main(argv):
>              f_out.write("  <Document>\n")
>              f_out.write("    <name>Location History</name>\n")
>              for item in items:
> +                if not dateCheck(item["timestampMs"]) :  continue
> Could be removed as per comment above.
> 
> In location_history_json_converter.py:
> 
> > @@ -149,8 +175,9 @@ def main(argv):
>                  # The deltas below assume input is in reverse chronological order.  If it's not, uncomment this:
>                  # items = sorted(data["data"]["items"], key=lambda x: x['timestampMs'], reverse=True)
>                  for item in items:
> +                    if not dateCheck(item["timestampMs"]) :  continue
> Could be removed as per comment above.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Scarygami commented:
@askariikeka apparently you decided to "watch" this repository a while back, which is why you get notifications when something happens: https://github.com/Scarygami/location-history-json-converter/watchers

You can unwatch the repository to stop receiving those mails.

rpeyron commented:
Hi, I made the changes you suggested (and some benchmarks that confirm these changes) ; I am not sure the PR has taken these changes : is it ok or should I make a new PR ?

rpeyron commented:
I forgot to mention in the log ; I also changed : 
- fixed the use of argv in arg_parser.parse_args   (and this is useful to use Timeit to bench)
- modifed the comment of gpxtracks order with the abs modification and --chronological new option

Scarygami commented:
Thanks!
